builtin/mainmenu/settings/dlg_settings.lua

```diff
@@ -463,6 +463,7 @@ local function buttonhandler(this, fields)
 	dialogdata.query = fields.search_query
 
 	if fields.back then
+		dialogdata.page_id = update_filtered_pages("")
```

Comment:
This won't work (iirc) if you press ESC.

Better to move this to create_settings_dlg

Reply:
Yes, sounds good.